File: app/src/main/java/com/redteamobile/smart/util/AssetManager.java
package com.redteamobile.smart.util;

import android.content.Context;
import android.content.Intent;

import androidx.localbroadcastmanager.content.LocalBroadcastManager;

import com.redteamobile.smart.Constant;

import java.io.File;

public class AssetManager {

    private static final String DEFAULT_ASSETS = "assets.der";
    private final String ASSETS = "rt_share_profile.der";
    private String cachePath;
    private String cacheFile;
    private Context context;

    public AssetManager(Context context) {
        super();
        cachePath = FileUtil.getAppPath(context) + "/";
        cacheFile = cachePath + ASSETS;
        this.context = context;
    }

    public void preloadedToCache() {
        File path = new File(this.cachePath);
        if (!path.exists()) {
            path.mkdirs();
        }
        File file = new File(this.cacheFile);
        if (file.exists()) {
            LocalBroadcastManager.getInstance(context)
                    .sendBroadcast(new Intent(Constant.ACTION_BOOTSTRAP_READY));
        }
        new AssetUtil(context).copyFileInThread(DEFAULT_ASSETS, ASSETS, cachePath);
    }
}

File: app/src/main/java/com/redteamobile/smart/agent/util/AssetManager.kt
package com.redteamobile.smart.agent.util

import android.content.Context
import android.content.Intent
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import com.redteamobile.smart.Constant
import java.io.File

class AssetManager {

    private val TAG: String = AssetManager::class.java.simpleName

    private val DEFAULT_ASSETS = "B200518021540844291_assets.der"
    private val ASSETS = "rt_share_profile.der"

    private var cachePath: String? = null
    private var cacheFile: String? = null
    private var context: Context

    constructor(context: Context) {
        this.context = context
        cachePath = FileUtil.getFilesDir(context).absolutePath.toString() + "/"
        cacheFile = cachePath + ASSETS
    }

    // todo 路径需要验证
    fun preloadedToCache() {
        LogUtil.i(TAG, "preloadedToCache(): $cachePath")
        val path = File(cachePath)
        if (!path.exists()) {
            path.mkdirs()
        }
        val file = File(cacheFile)
        if (file.exists()) {
            context?.let { LocalBroadcastManager.getInstance(it).sendBroadcast(Intent(Constant.ACTION_BOOTSTRAP_READY)) }
        }
        AssetUtil(context).copyFileInThread(DEFAULT_ASSETS, ASSETS, cachePath)
    }
}
